lua/starfall/libs_sh/vehicles.lua

```diff
@@ -1,18 +1,57 @@
 -- Global to all starfalls
 local checkluatype = SF.CheckLuaType
 local registerprivilege = SF.Permissions.registerPrivilege
-local ENT_META = FindMetaTable("Entity")
-local PLY_META = FindMetaTable("Player")
-local VEH_META = FindMetaTable("Vehicle")
+local ENT_META,PLY_META,VEH_META = FindMetaTable("Entity"),FindMetaTable("Player"),FindMetaTable("Vehicle")
 
+
+local UseEnableVehicles
 if SERVER then
 	-- Register privileges
 	registerprivilege("vehicle.eject", "Vehicle eject", "Removes a driver from vehicle", { entities = {} })
 	registerprivilege("vehicle.kill", "Vehicle kill", "Kills a driver in vehicle", { entities = {} })
 	registerprivilege("vehicle.strip", "Vehicle strip", "Strips weapons from a driver in vehicle", { entities = {} })
 	registerprivilege("vehicle.lock", "Vehicle lock", "Allow vehicle locking/unlocking", { entities = {} })
-end
 
+	local sf_max_driveruse_dist = CreateConVar("sf_vehicle_use_distance", 100, FCVAR_ARCHIVE, "The max reach distance allowed for Vehicle:driverUse function.")
+
+	local Ent_IsValid = ENT_META.IsValid
+	local Ply_GetVehicle,Ply_GetEyeTrace = PLY_META.GetVehicle,PLY_META.GetEyeTrace
+
+	UseEnableVehicles = {
+		setEnabled = function(self, vehicle, enabled)
+			if enabled then
+				self:addhooks()
+				self.vehicles[vehicle] = true
+			else
+				self.vehicles[vehicle] = nil
+				self:removehooks()
+			end
+		end,
+		use = function(self, ply, veh)
+			local tr = Ply_GetEyeTrace(ply)
+			local ent = tr.Entity
+			if Ent_IsValid(ent) and tr.HitPos:DistToSqr(tr.StartPos) <= sf_max_driveruse_dist:GetFloat()^2 and not ent:IsVehicle() and ent.Use and hook.Run("PlayerUse", ply, ent) ~= false then
+				ent:Use(ply, veh, USE_ON, 0)
+			end
+		end,
+		addhooks = function(self)
+			if table.IsEmpty(self.vehicles) then
+				hook.Add("KeyPress","SF_VehicleButtons",function(ply, key)
+					if key==IN_ATTACK then
```

Comment:
The only minor complaint I have about this, it is not good to have hard-coded to attack key (add a client-side userinfo convar to allow players to customize their "use" key). Then it is good.

Reply:
Eh, a cvar is kinda shit, maybe better to add an arg to let the SF function control what key triggers it.

Reply:
Na, nvm that'd be too confusing. It should just always be the attack input. Users can bind attack to whatever they want.